metatensor-torch/src/atomistic.cpp

```diff
@@ -0,0 +1,986 @@
+#include <cctype>
+#include <cstring>
+
+#include <array>
+#include <sstream>
+#include <algorithm>
+
+#include <torch/torch.h>
+#include <nlohmann/json.hpp>
```

Comment:
I'm worried about the compilation time with this new dependency. I assume that this header includes a lot of smaller headers; can we get the same functionality using one of the smaller headers?

Reply:
It did not felt to me a lot slower to compile here. In practise, I find that torch headers contain a lot more code and are the main part slowing down compilation here. I'll run some compile-time tests here.

We could use something smaller like https://github.com/DaveGamble/cJSON, but it's a pure C library, so the code is going to get a lot more convoluted.

Reply:
Ok... I just assumed torch wasn't a problem because it's compiled, but I guess it's very large